bindings/ergo-lib-wasm/tests_browser/test_rest_api.js

```diff
@@ -10,8 +10,32 @@ beforeEach(async () => {
 // particular the timeout functionality for HTTP requests requires the window object from the
 // web APIs, thus requiring a web browser to run.
 
+it('node REST API: peer_discovery endpoint', async () => {
+    console.log("Starting peer discovery");
+    const seeds = get_ergo_node_seeds();
+
+    // Limit to 150 simultaneous HTTP requests and search for peers for 10 seconds (remember
+    // there's an unavoidable waiting time of 80 seconds, to give Chrome time to relinquish failed
+    // preflight requests)
+    let res = await ergo_wasm.peer_discovery_chrome(seeds, 150, 220);
+    assert(res.len() > 0, "Should be at least one peer!");
+});
+
+it('node REST API: peer_discovery endpoint (INCREMENTAL VERSION)', async () => {
+    const seeds = get_ergo_node_seeds();
+    let scan = new ergo_wasm.ChromePeerDiscoveryScan(seeds);
+
+    scan = await ergo_wasm.incremental_peer_discovery_chrome(scan, 150, 90);
+    let scan_1_len = scan.active_peers().len();
+    scan = await ergo_wasm.incremental_peer_discovery_chrome(scan, 150, 400);
+    let scan_2_len = scan.active_peers().len();
+    assert(scan_1_len < scan_2_len, "Should have found more peers after second scan!");
+});
+
 it('node REST API: get_nipopow_proof_by_header_id endpoint', async () => {
-    let node_conf = new ergo_wasm.NodeConf("213.239.193.208:9053");
+    let peers = await ergo_wasm.peer_discovery_chrome(get_ergo_node_seeds(), 150, 150);
```

Comment:
I'm not sure we should use live peer discovery in tests. Only in tests of the peer discovery itself (two tests above). In all other tests, a hardcoded node list should suffice.

Reply:
Yeah, that's not good. However, running peer discovery in these tests looks like a huge overkill. Plus, I'm not sure how github views if we shoot a lot of network requests in GA. It might be a violation of their TOS.
What do you think of running a peer discovery once per test session and making discovered peers list accessible in relevant tests?

Reply:
That's a great point concerning the TOS. I've implemented what you've suggested and it's surfaced an issue specific to github CI where many active peers aren't getting picked up compared with running it locally. I think it's due to the timeout interval of each individual request (it's currently set to 6 seconds). You can see I've made a bunch of noisy commits where I'm experimenting, so sorry about that. I hope to resolve this soon!

Reply:
I just did a rough test, and my guess doesn't seem right. With the incremental test, the table below shows the number of active nodes picked up by Github and locally, for each incremental round:
|   | Github Actions| Locally|
|---|---|--|
|increment 1| 0| 6|
| increment 2| 2 | 21|

So there's something going on with Github actions. Is it ok for me to clean things up, set hardcoded IPs again for the SPV test and leave it here? Then I can finally focus on the oracle-core tasks : )

Reply:
Sure. No need to dig further with CI.

Reply:
Thanks. Please note the current error in CI, where the 2nd incremental round found no new peers. It's unfortunate, as it never happens locally.

Reply:
After latest commit, error is in the same place. I made issue #586 for this. As a temporary measure, we can allow the 2nd round to find no new peers, just to silence the CI errors that will crop up from everyone's commits. What do you think?

Reply:
Actually, it looks like the problem comes about from persisting `peer_discovery` results between tests. Reverted this, and `peer_discovery` starts finding more active nodes. Looks like this is fixable. Never a boring day with web dev...

Reply:
It's still nowhere near the performance of running it locally though:
<img width="654" alt="image" src="https://user-images.githubusercontent.com/791016/176726741-b6106bbf-3b17-4af8-8a7e-6d5403b75dff.png">

So a single round of `peer_discovery` with search time 60 seconds yields 3 active nodes. For the incremental test, the first round is 10 second search time, and the second round is 400 seconds search time. The second round only finds one extra node, which isn't good at all.

But I think everything's ready for merging though!

Reply:
Awesome!